Source/WebKit/Shared/ApplePay/ios/WebPaymentCoordinatorProxyIOS.mm

```diff
@@ -71,14 +71,16 @@
             return completionHandler(false);
 
 #if ENABLE(APPLE_PAY_REMOTE_UI_USES_SCENE)
-        paymentCoordinatorProxy->checkedClient()->getWindowSceneAndBundleIdentifierForPaymentPresentation(webPageProxyID, [weakThis = WTFMove(weakThis), completionHandler = WTFMove(completionHandler)](const String& sceneIdentifier, const String& bundleIdentifier) mutable {
+        paymentCoordinatorProxy->checkedClient()->getWindowSceneAndBundleIdentifierForPaymentPresentation(webPageProxyID, [weakThis = WTFMove(weakThis), completionHandler = WTFMove(completionHandler), webPageProxyID](const String& sceneIdentifier, const String& bundleIdentifier) mutable {
             auto paymentCoordinatorProxy = weakThis.get();
             if (!paymentCoordinatorProxy)
                 return completionHandler(false);
 
             if (!paymentCoordinatorProxy->m_authorizationPresenter)
                 return completionHandler(false);
 
+            paymentCoordinatorProxy->checkedClient()->notifyWillPresentPaymentUI(webPageProxyID);
+
             paymentCoordinatorProxy->m_authorizationPresenter->presentInScene(sceneIdentifier, bundleIdentifier, WTFMove(completionHandler));
```

Comment:
`notifyWillPresentPaymentUI` is an async IPC message, so we no longer have a guarantee on the call ordering between `willPresentModalUI()` and `presentInScene()`. I think it'll be wrong for us to present and then notify the client, right?

You probably want to make `notifyWillPresentPaymentUI` asynchronously call a completion handler after it notifies the client.

i.e.

```cpp
void NetworkConnectionToWebProcess::notifyWillPresentPaymentUI(WebPageProxyIdentifier webPageProxyIdentifier, CompletionHandler<void(void)>&& completionHandler)
{
    networkProcess().parentProcessConnection()->send(Messages::NetworkProcessProxy::NotifyWillPresentPaymentUI(webPageProxyIdentifier), WTFMove(completionHandler));
}
```